userprofile/templates/userprofile/login.html

```diff
@@ -1,99 +1,112 @@
 {% extends "base.html" %}
 {% load i18n %}
+{% load static %}
 {% load bootstrap %}
 
 {% block title %}{% trans "Login" %} | {{ block.super }}{% endblock %}
 
+{% block scripts %}
+	{{ block.super }}
+	<link rel="stylesheet" href="{% static 'userprofile/login.css' %}" />
+	<script src="{% static 'userprofile/extra_logins.js' %}"></script>
+{% endblock %}
+
 {% block content %}
 
-<div class="page-header">
-    <h1>{% trans "Login" %}</h1>
-</div>
+	<div class="page-header">
+		<h1>{% trans "Login" %}</h1>
+	</div>
+
+	<div id="login-box-row" class="row">
 
-<div class="row">
+		{% if shibboleth_login %}
+			<div class="col-sm-6 col-md-3 login-box">
+				<div class="thumbnail">
+					<div class="caption">
+						{% if shibboleth_title_text %}
+							<h3>
+								{{ shibboleth_title_text|safe }}
+							</h3>
+						{% endif %}
+						{% if shibboleth_body_text %}
+							<p>
+								{{ shibboleth_body_text|safe }}
+							</p>
+						{% endif %}
+						<p>
+							<a class="btn btn-primary btn-block" role="button" href="{% url 'shibboleth-login' %}?next={{ next|iriencode }}">
+								{{ shibboleth_button_text|safe }}
+							</a>
+						</p>
+					</div>
+				</div>
+			</div>
+		{% endif %}
 
```

Comment:
In general however, logging in as a teacher and student seem to work well with the currently available "Local user"-log in.